image/image_linux.go

```diff
@@ -481,28 +482,38 @@ func (s *imageSeeder) deriveInfoForLocalSnaps(f seedwriter.SeedAssertionFetcher,
 	return snaps, s.w.InfoDerived()
 }
 
-func (s *imageSeeder) validationSetKeys() ([]snapasserts.ValidationSetKey, error) {
+func (s *imageSeeder) validationSetsAndRevisionForSnap(snapName string) ([]snapasserts.ValidationSetKey, snap.Revision, error) {
 	vsas, err := s.db.FindMany(asserts.ValidationSetType, nil)
 	if err != nil && !errors.Is(err, &asserts.NotFoundError{}) {
-		return nil, err
+		return nil, snap.Revision{}, err
 	}
 
-	var vsKeys []snapasserts.ValidationSetKey
+	allVss := snapasserts.NewValidationSets()
 	for _, a := range vsas {
-		vsa := a.(*asserts.ValidationSet)
-		vsKeys = append(vsKeys, snapasserts.NewValidationSetKey(vsa))
+		if err := allVss.Add(a.(*asserts.ValidationSet)); err != nil {
+			return nil, snap.Revision{}, err
+		}
+	}
+
```

Comment:
it feels like we should run at least a Conflict check here on allVss

Reply:
Added this in, along with a unit test with conflicting validation-sets